File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/domain/Student.java
package org.jimmer.jimmerdatasouce.domain;

import org.babyfish.jimmer.Formula;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "db_student")
public interface Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @Key
    String firstName();

    @Key
    String lastName();

    @IdView
    @Nullable
    Long collegeId();

    Gender gender();

    @Nullable
    LocalDateTime birthday();

    /**
     * 简单计算属性，获取全名
     */
    @Formula(dependencies = {"firstName", "lastName"})
    default String fullName() {
        return firstName() + lastName();
    }

    /**
     * 多对多关系映射 主动方
     * 由于本项目基于中间表关联，数据库层面没有真正的外键，所以使用 joinColumns、inverseJoinColumns
     * 如果是真外键，使用 joinColumnName、inverseJoinColumnName 指定主键名称即可
     * 使用 @JoinTable (name = "中间表名称") 指定中间表名称，而不是使用默认的 两表类名_MAPPING(STUDENT_COURSE_MAPPING)
     */
    @ManyToMany
    @JoinTable(
            name = "db_stu_course",
            joinColumns = {@JoinColumn(name = "stu_id")},
            inverseJoinColumns = {@JoinColumn(name = "course_id")}
    )
    List<Course> courses();

    @ManyToOne
    @Nullable
    College college();
}

File: jimmer-example/micronaut-with-jimmer-kt/src/main/kotlin/com/example/domain/Student.kt
package com.example.domain;

import org.babyfish.jimmer.Formula
import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 学生信息
 * @author jhlz
 * @since 2024/4/16 9:54
 * @version x.x.x
 */
@Entity
@Table(name = "db_student")
interface Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Key
    val firstName: String

    @Key
    val lastName: String

    val collegeId: Long?

    val gender: String

    /**
     * 简单计算属性，获取全名
     */
    @Formula(dependencies = ["firstName", "lastName"])
    val fullName: String get() = "$firstName$lastName"

    val birthday: LocalDateTime?

    @LogicalDeleted("2")
    @Column(name = "is_delete")
    val delete: Int

}